airbyte-webapp/src/components/connectorBuilder/Builder/AddStreamButton.tsx

```diff
@@ -0,0 +1,92 @@
+import classNames from "classnames";
+import { Form, Formik, useFormikContext } from "formik";
+import { useState } from "react";
+import { FormattedMessage } from "react-intl";
+
+import { Button } from "components/ui/Button";
+import { Modal, ModalBody, ModalFooter } from "components/ui/Modal";
+
+import { ReactComponent as PlusIcon } from "../../connection/ConnectionOnboarding/plusIcon.svg";
+import styles from "./AddStreamButton.module.scss";
+import { FormikPatch } from "./Builder";
+import { BuilderField } from "./BuilderField";
+
+interface AddStreamValues {
+  streamName: string;
+  urlPath: string;
+}
+
+interface AddStreamButtonProps {
+  className?: string;
+  numStreams: number;
+  onAddStream: (addedStreamNum: number, addedStreamName: string) => void;
+}
+
+export const AddStreamButton: React.FC<AddStreamButtonProps> = ({ className, numStreams, onAddStream }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const { setFieldValue } = useFormikContext();
+
+  return (
+    <>
+      <Button
+        className={classNames(className, styles.addButton)}
+        onClick={() => {
+          setIsOpen(true);
+        }}
+        icon={<PlusIcon className={styles.plus} />}
+      />
+      {isOpen && (
+        <Formik
+          initialValues={{ streamName: "", urlPath: "" }}
+          onSubmit={(values: AddStreamValues) => {
+            setFieldValue(`streams[${numStreams}]`, {
```

Comment:
nit: I'm not 100% sure whether it's a problem, but it looks like a code smell to me that adding a new item to an array is done by setting a numeric property of the next index. The more idiomatic way seems to set the field value of `streams` and spread the existing array of streams and adding the new one:
```
setFieldValue("streams", [...oldStreams, {
name: values.streamName,
              urlPath: values.urlPath,
              fieldPointer: [],
              httpMethod: "GET",
}])
```

Feel free to disregard

Reply:
I think this makes a lot of sense, I made this change and it allowed me to simplify a few different components 👍 thanks for the suggestion!